storage/types.go

```diff
@@ -0,0 +1,41 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/TykTechnologies/tyk/config"
+	"github.com/TykTechnologies/tyk/storage/internal"
+)
+
+// Controller interface for storage controllers
+type Controller interface {
+	Connect(ctx context.Context, onReconnect func(), conf *config.Config)
+	Context() context.Context
+	Connected() bool
+	Disable(setRedisDown bool)
+	WaitConnect(ctx context.Context) bool
+
+	ControllerDeprecated
+}
+
+// ControllerDeprecated are old API endpoints for backwards compatibility
+type ControllerDeprecated interface {
```

Comment:
should we deprecate this directly and makes the necessary code changes?

Reply:
No because It will break our tightly coupled CI builds (integration tests with dashboard). We need to correct usage where appropriate before we delete this. We can do whatever, but the first step is keeping all BC.